### PR TITLE
[eclipse/xtext-maven#38] Switch to annotations instead of doclet

### PR DESCRIPTION
**Pull request by jrosczak**
Replace the special maven javadoc annotations to describe the api of
this plugin with real maven annotations.

Signed-off-by: Jan Rosczak <jan.rosczak@web.de>

### CODE REVIEW

File: org.eclipse.xtext.maven.plugin/pom.xml [hunk 1]
```diff
@@ -9,9 +9,7 @@
 	<modelVersion>4.0.0</modelVersion>
 	<!-- groupId and version must be repeated for publishing
 		see https://github.com/eclipse/xtext-maven/pull/40#issuecomment-399433375 -->
-	<groupId>org.eclipse.xtext</groupId>
```
Comment (cdietrich) [inline org.eclipse.xtext.maven.plugin/pom.xml]: please keep this dips. we need it for gradle publish, see comment above

File: org.eclipse.xtext.maven.plugin/pom.xml [hunk 2]
```diff
@@ -41,6 +39,12 @@
 			<version>${maven.version}</version>
 			<scope>provided</scope>
 		</dependency>
+        <dependency>
```
Comment (szarnekow) [inline org.eclipse.xtext.maven.plugin/pom.xml]: Minor: Please use tabs for indentation.

**Comment by cdietrich** [comment]
From my POV ok, what do you think @kthoms @szarnekow

**Comment by cdietrich** [comment]
thx @jrosczak

**Pull request merged**